sympy/polys/matrices/domainmatrix.py

```diff
@@ -30,6 +30,15 @@ def __init__(self, rows, shape, domain):
         self.shape = shape
         self.domain = domain
 
+    def __getitem__(self, key):
+        return self.rep.to_list()[key]
```

Comment:
Converting the whole matrix to a list just to extract one element is obviously not efficient.